docs/style.md

```diff
@@ -93,3 +93,61 @@ While convenient for editing, it may be poorly legible in unrendered form
 <!-- BAD -->
 Definitely don't use soft-wrapping. While markdown mostly ignores source level line breaks, relying on soft wrap makes the source completely unreadable, especially on modern wide displays.
 ```
+
+## Code guidelines
+
+- Big changes in Pull Requests
+
+When making Pull Requests it's important to try to create PR in such a way
+that minimizes changes of breaking the codebase, and making it easy to review.
+It's may be near impossible to define strict guidelines, but to make PR
+betters let's try to avoid common mistakes.
+
+```markdown
+<!-- BAD -->
+Making a 70 files change, where 67 files changed are because of refactoring,
+and 3 actual code changes. Let's try to keep big refactorings separate from the
+rest of code.
```

Comment:
maybe also mention why this is bad:

In this scenario, reviewer will have to do the work to distinguish which 3 files contain the real change, vs the other 67 that have 'just a refactoring'.